test/with_dummyserver/test_socketlevel.py

```diff
@@ -1399,7 +1399,7 @@ class TestRetryPoolSizeDrainFail(SocketDummyServerTestCase):
 
     def test_pool_size_retry_drain_fail(self):
         def socket_handler(listener):
-            while 1:
+            for _ in xrange(2):
```

Comment:
`xrange` isn't present on Python 3. We don't mind an intermediate list being created on Py 2 because it's only two elements long, so regular `range` is fine.